src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -176,7 +176,12 @@ public void run() {
 
     private boolean createJenkinsHook(GHRepository repo, URL url) {
         try {
-            repo.createHook("jenkins", Collections.singletonMap("jenkins_hook_url", url.toExternalForm()), null, true);
+            if (getDescriptor().isUseWebHooks()) {
+                // No need to pass events since the default event is PUSH
+                repo.createWebHook(new URL(url.toExternalForm()));
+            } else {
+                repo.createHook("jenkins", Collections.singletonMap("jenkins_hook_url", url.toExternalForm()), null, true);
```

Comment:
The main and most important difference is that you're allowed to register more than one Jenkins instance on the same repository, something that using the "jenkins" hook does not allow.